presto-hive/src/main/java/com/facebook/presto/hive/cache/HiveCachingHdfsConfiguration.java

```diff
@@ -62,10 +62,11 @@ public HiveCachingHdfsConfiguration(
     @Override
     public Configuration getConfiguration(HdfsContext context, URI uri)
     {
+        Configuration defaultConfig = hiveHdfsConfiguration.getConfiguration(context, uri);
         @SuppressWarnings("resource")
         Configuration config = new CachingJobConf((factoryConfig, factoryUri) -> {
             try {
-                FileSystem fileSystem = (new Path(factoryUri)).getFileSystem(hiveHdfsConfiguration.getConfiguration(context, factoryUri));
+                FileSystem fileSystem = (new Path(factoryUri)).getFileSystem(defaultConfig);
```

Comment:
If different uris could yield different configuration, then this change is incorrect. The previous logic uses `factoriUri` to get configuration, the new one uses `uri`. If this two are equivalent, then there's no need to have two configs.

Reply:
Good catch. Will add a check to see if the URI has changed since the creation of the JobConf object. If the URI did not change we can avoid the extra call to `getConfiguration()`.

Reply:
If the URI does not change why do we need a factory method here?

Reply:
the URI might change in the workers.